### PR TITLE
[6.4] UI airgun port sync status tests

### PR DESCRIPTION
**Pull request by ldjebran**
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_sync.py 
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.2.1, mock-1.10.0, forked-0.2, cov-2.5.1
collecting 4 items                                                                                           2018-09-24 18:42:32 - conftest - DEBUG - BZ deselect is disabled in settings

collected 4 items                                                                                            

tests/foreman/ui_airgun/test_sync.py::test_positive_sync_custom_repo PASSED                            [ 25%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_rh_repos PASSED                               [ 50%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_custom_ostree_repo PASSED                     [ 75%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_rh_ostree_repo PASSED                         [100%]

========================================= 4 passed in 328.67 seconds =========================================
```
depend on https://github.com/SatelliteQE/airgun/pull/209

### CODE REVIEW

File: tests/foreman/ui_airgun/test_sync.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    ).create()
+    with session:
+        session.organization.select(org_name=module_org.name)
+        results = session.sync_status.synchronize([
+            (module_custom_product.name, repo.name)])
+        assert len(results) == 1
+        assert results[0] == 'Syncing Complete.'
+
+
+@run_only_on('sat')
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_sync.py]: +

File: tests/foreman/ui_airgun/test_sync.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        RHELCloudFormsTools(cdn=True)
+    )
+    distros = [DISTRO_RHEL7, DISTRO_RHEL6]
+    repo_collections = [
+        RepositoryCollection(distro=distro, repositories=[repo])
+        for distro, repo in zip(distros, repos)
+    ]
+    for repo_collection in repo_collections:
+        repo_collection.setup(module_org.id, synchronize=False)
+    repo_paths = []
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_sync.py]: I am not sure that I like that implementation or one that we have in old robottelo, but appreciate if you can make it more generic and smarter without all these `if` conditions and etc

**Comment by ldjebran** [comment]
@oshtaier comments addressed
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_sync.py 
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.3.0, mock-1.10.0, forked-0.2, cov-2.5.1
collecting 4 items
2018-10-08 12:58:42 - conftest - DEBUG - BZ deselect is disabled in settings

collected 4 items                                                                                            

tests/foreman/ui_airgun/test_sync.py::test_positive_sync_custom_repo PASSED                            [ 25%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_rh_repos PASSED                               [ 50%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_custom_ostree_repo PASSED                     [ 75%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_rh_ostree_repo PASSED                         [100%]

========================================= 4 passed in 279.45 seconds =========================================
```

**Comment by ldjebran** [comment]
remove do not merge as dependency merged

**Comment by oshtaier** [comment]
ACK pending conflict resolve

**Comment by ldjebran** [comment]
@oshtaier rebased and resolved conflicts
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$  pytest -v tests/foreman/ui_airgun/test_sync.py 
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.3.0, mock-1.10.0, forked-0.2, cov-2.5.1
collecting 4 items                                                                                           2018-10-09 09:19:37 - conftest - DEBUG - BZ deselect is disabled in settings

collected 4 items                                                                                            

tests/foreman/ui_airgun/test_sync.py::test_positive_sync_custom_repo PASSED                            [ 25%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_rh_repos PASSED                               [ 50%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_custom_ostree_repo PASSED                     [ 75%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_rh_ostree_repo PASSED                         [100%]

======================================== 4 passed in 1102.03 seconds =========================================
```

File: tests/foreman/ui_airgun/test_sync.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+
+@fixture(scope='module')
+def module_custom_product(module_org):
+    return entities.Product(organization=module_org).create()
+
+
+@fixture(scope='module')
+def org_manifest(module_org):
+    manifests.upload_manifest_locked(module_org.id)
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_sync.py]: redundant as such fixture won't be used anywhere. Pls add this step to `module_org` instead.

File: tests/foreman/ui_airgun/test_sync.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    with session:
+        session.organization.select(org_name=module_org.name)
+        results = session.sync_status.synchronize([
+            (module_custom_product.name, repo.name)])
+        assert len(results) == 1
+        assert results[0] == 'Syncing Complete.'
+
+
+@run_in_one_thread
+@skip_if_not_set('fake_manifest')
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_sync.py]: This check doesn't make much sense as fake manifest is used for module org, meaning all the tests are dependent on fake_manifest. Can we apply it to all the tests in the module somehow? Or maybe use 2 different fixtures (module_org & org_with_manifest)?

File: tests/foreman/ui_airgun/test_sync.py (hunk 3 above)
Comment (ldjebran) [reply tests/foreman/ui_airgun/test_sync.py]: @abalakh I am not sure about this as some tests are running in one thread and use manifest, others run in parallel and do not use manifest.

File: tests/foreman/ui_airgun/test_sync.py (hunk 4 above)
Comment (ldjebran) [reply tests/foreman/ui_airgun/test_sync.py]: @abalakh we have 2 tests that will run in parallel and use manifest 2 other tests run in thread and use manifest

File: tests/foreman/ui_airgun/test_sync.py (hunk 3 above)
Comment (abalakh) [reply tests/foreman/ui_airgun/test_sync.py]: But you're updating the same organization, so module_org will receive manifest anyways.

Comment (ldjebran) [reply tests/foreman/ui_airgun/test_sync.py]: no , the other tests are running not in one thread and will not need manifest

**Comment by ldjebran** [comment]
@abalakh @oshtaier changes done
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_sync.py 
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.3.0, mock-1.10.0, forked-0.2, cov-2.5.1
collecting 4 items                                                                                           2018-10-12 16:57:17 - conftest - DEBUG - BZ deselect is disabled in settings

collected 4 items                                                                                            

tests/foreman/ui_airgun/test_sync.py::test_positive_sync_custom_repo PASSED                            [ 25%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_rh_repos PASSED                               [ 50%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_custom_ostree_repo PASSED                     [ 75%]
tests/foreman/ui_airgun/test_sync.py::test_positive_sync_rh_ostree_repo PASSED                         [100%]

======================================== 4 passed in 1550.40 seconds =========================================
```

**Pull request merged**